release notes/v0.30.0.md

```diff
@@ -0,0 +1,123 @@
+
+And yes, you can have multiple `SharedArray` objects in the same script, just make sure to give them unique names - this is what `someName` in the script above was for. Because VUs are independent JS runtimes, we need some way to differentiate between the different shared memory objects, so we require them to have unique names. These names are also the IDs that any [xk6](https://github.com/k6io/xk6) extensions would need to use to access them.
+
+This works both locally and in the cloud. We advise everyone that deals with huge data files to wrap them in a `SharedArray` and give this new feature a try. The required script changes should be minimal, while the benefits should be significant. Hopefully, we can finally consider one of the biggest blockers k6 users have had for a long time solved! :tada:
+
+### Support a `handleSummary()` callback at the end of the test ([#1768](https://github.com/loadimpact/k6/pull/1768))
+
+You can now `export` a function called `handleSummary()` and k6 will call it at the end of the test run, after even `teardown()`. `handleSummary()` will be called with a JS object containing the same information that is used to generate the end-of-test summary and `--summary-export`, and allows users to completely customize how the end-of-test summary looks like.
+
+Besides customizing the end-of-test CLI summary (if `handleSummary()` is exported, k6 will not print the default), you can also generate and save as files various machine-readable data formats with the same data (JSON, CVS, JUnit/xUnit/etc. XML, etc.), as well as human-readable formats like HTML. Even binary formats like PDF are not out of reach, potentially, with an appropriate JS library that works in k6! You can also send the generated reports to a remote server by making an HTTP request with them (or using any of the other protocols k6 already supports)!
+
+k6 expects `handleSummary()` to return a `{key1: value1, key2: value2, ...}` map. The values can be string or `ArrayBuffer` and represent the generated summary report contents. The keys should be strings and determine where the contents will be displayed or saved: `stdout` for [standard output](https://en.wikipedia.org/wiki/Standard_streams), `stderr` for standard error, or a path to a file on the system (which will be overwritten). Here's a simple example:
```

Comment:
Maybe the above explanation of the possible uses and especially the listing of the formats should be after this example. Although then it will be harder for someone just skimming this to notice that they can now maybe generate their custom JSON report directly in k6. 

So maybe someone with better English than mine should try to restructure it even more :thinking:

Reply:
hmm this paragraph could be moved after the example, but I'd prefer the previous paragraph to stay before

Reply:
see if https://github.com/loadimpact/k6/pull/1809/commits/80ca8a30809819556f409fdf55e8b6a06fbbb306 is better?